jmclient/jmclient/taker.py

```diff
@@ -579,40 +587,35 @@ def on_sig(self, nick, sigb64):
                 break
             if len(sig_deserialized) == 2:
                 ver_sig, ver_pub = sig_deserialized
-                scriptCode = None
             elif len(sig_deserialized) == 3:
-                ver_sig, ver_pub, scriptCode =  sig_deserialized
+                ver_sig, ver_pub, trash = sig_deserialized
```

Comment:
This is interesting. First, it's traditional to use `_` for unused return values, like `a, b, _ = func_that_returns_a_tuple()` ... I suspect for linting reasons it's preferable, though not important.

That we formerly had that third item is interesting - it was mostly being used as a flag - thus the maker is clearly identifying he is providing a wrapped segwit signature - but secondarily as a sanity check on the redeemScript. But of course it technically wasn't necessary for that secondary reason because the pubkey is provided and you could reconstruct it from that; even though that represents an assumption, it's a fine one to make (anything else is unrecognized, which is already true for anything unexpected).

Here you've used the python-bitcointx functions to make the processing cleaner - take the sig and pub data and check the type of the scriptPubKey, then process it based on the assumption that it is one of those three types (so as before it assumes that any p2sh is p2sh-p2wpkh and fails otherwise.

There is an argument of course (a good one really) for generalising further so that we can accept any old inputs. It complicates the bot messaging protocol but we probably should have done it from day one, it's just our bitcoin backend library wasn't powerful enough (or safe enough I guess); we can have the bots just send full blown PSBTs (modulo the bandwidth, ouch), or just stripped down: scriptSig and scriptPubKey in bitcoin serialization. But that's for another day.

I think what you have in this function is good.
Second, I believe you've got all the logic right.